ToDoLite/src/main/java/com/couchbase/todolite/Application.java

```diff
@@ -95,148 +112,147 @@ private synchronized void updateSyncProgress(int completedCount, int totalCount,
-
-        }
+        sync = new Synchronize.Builder(getDatabase(), SYNC_URL)
+                .addChangeListener(getReplicationChangeListener())
+                .build();
+        sync.start();
 
     }
 
     public void startReplicationSyncWithStoredBasicAuth() {
 
-        if (pullReplication == null && pushReplication == null) {
-
-            Replication[] replications = createReplications();
-            pullReplication = replications[0];
-            pushReplication = replications[1];
+        sync = new Synchronize.Builder(getDatabase(), SYNC_URL)
+                .basicAuth(getCurrentUserId(), getCurrentUserPassword())
+                .addChangeListener(getReplicationChangeListener())
+                .build();
+        sync.start();
 
-            String password = getCurrentUserPassword();
-            if (password == null || password.isEmpty()) {
-                Log.w(TAG, "Cannot start replication, no saved username / password");
-                return;
-            }
-            String username = getCurrentUserId();
+    }
 
-            Authenticator basicAuthenticator = AuthenticatorFactory.createBasicAuthenticator(username, password);
 
-            pullReplication.setAuthenticator(basicAuthenticator);
-            pushReplication.setAuthenticator(basicAuthenticator);
+    public void startReplicationSyncWithBasicAuth(String username, String password) {
 
-            pullReplication.start();
-            pushReplication.start();
+        sync = new Synchronize.Builder(getDatabase(), SYNC_URL)
+                .basicAuth(username, password)
+                .addChangeListener(getReplicationChangeListener())
+                .build();
+        sync.start();
 
-            Log.v(TAG, "startReplicationSyncWithStoredCustomCookie(): Start Replication Sync ...");
+    }
 
-        } else {
-            Log.v(TAG, "startReplicationSyncWithStoredCustomCookie(): doing nothing, already have existing replications");
+    public void startReplicationSyncWithFacebookLogin(String accessToken) {
 
-        }
+        sync = new Synchronize.Builder(getDatabase(), SYNC_URL)
+                .facebookAuth(accessToken)
+                .addChangeListener(getReplicationChangeListener())
+                .build();
+        sync.start();
 
     }
 
+    public void migrateGuestDataToUser(Document profile) {
+        Database guestDb = getDatabaseForGuest();
+        if (guestDb.getLastSequenceNumber() > 0) {
 
-    public void startReplicationSyncWithBasicAuth(String username, String password) {
+            QueryEnumerator rows = null;
+            try {
+                rows = guestDb.createAllDocumentsQuery().run();
+            } catch (CouchbaseLiteException e) {
+                Log.e(TAG, "Cannot run all docs query", e);
+            }
 
-        if (pullReplication == null && pushReplication == null) {
+            if (rows.getCount() == 0) {
```

Comment:
Nothing wrong with this but the code might be cleaner with just calling guestDb.getDocumentCount().